pkg/compact/compact.go

```diff
@@ -284,10 +283,6 @@ func NewDefaultGrouper(
 			Name: "thanos_compact_group_vertical_compactions_total",
 			Help: "Total number of group compaction attempts that resulted in a new block based on overlapping blocks.",
 		}, []string{"resolution"}),
-		overlappingBlocks: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
```

Comment:
move this metric to its own source file, this will avoid conflicts when merge from upstream.